proxy/src/main.rs

```diff
@@ -95,6 +108,8 @@ async fn main() -> anyhow::Result<()> {
     let config: &ProxyConfig = Box::leak(Box::new(ProxyConfig {
         tls_config,
         auth_backend,
+        metric_collection_endpoint,
+        metric_collection_interval,
```

Comment:
There's a tls config and an auth backend config, which leads me to think that there should be metric collection config containing those two fields.

I'm not into prefixes.